java-frontend/src/main/java/org/sonar/java/bytecode/se/BytecodeEGWalker.java

```diff
@@ -323,21 +376,28 @@ void executeInstruction(Instruction instruction) {
         programState = programState.put(index, sv).addConstraint(sv, ObjectConstraint.NOT_NULL);
         break;
       case I2L:
-      case I2F:
       case I2D:
-      case L2I:
-      case L2F:
-      case L2D:
-      case F2I:
       case F2L:
       case F2D:
+        sv = programState.peekValue();
+        Preconditions.checkNotNull(sv, "%s needs value on stack", instruction.opcode);
```

Comment:
This is going to display the numerical value, and not the real code, right?

Reply:
fixed by relying on `Instruction.toString()` , also on few other places.